### ISSUE TITLE
Opening menu in popout window throws exception

### ISSUE THREAD

**Issue by fporzecki** [szhsin/react-menu#910]
React/React-dom version: 18.0.0
React-Menu version: 3.5.2

## Describe the bug
I've created a menu in a new window created by window.open() function with the use of React.Portal. Exact implementation can be found here: https://blog.scottlogic.com/2019/10/29/popout-windows-in-react.html

When trying to open the menu in this new window the app crashes with an error: "Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'."

From a brief check it seems that the issue lies in getScrollAncestor function in utils.js, more explanations in Code snippets section. Let me know should you need any other details.

## Expected behavior
No error should be thrown and menu should open without any issues.

## Code snippets/CodeSandbox examples/Screenshots
So here's the original getScrollAncestor function:
```
export const getScrollAncestor = (node) => {
  while (node) {
    node = node.parentNode;
    if (!node || node === document.body) return;
    const { overflow, overflowX, overflowY } = getComputedStyle(node);
    if (/auto|scroll|overlay|hidden/.test(overflow + overflowY + overflowX)) return node;
  }
};
```
And the problem lies in
```
if (!node || node === document.body) return;
```
Since our node exists in a new window we do go through all of its parents and finally reach the root where we should compare our node to document.body but document.body and node are not equal. That causes object of type different than Element to go to getComputedStyle, which in turn throws an exception.

A quick check from a forked repo shows that a change of condition to
```
if (!node || node === document.body || !node.parentNode) return;
```
where we check if we have no other parents (we've reached our 'root') does the job and with that I was able to open the menu in a new window.

**Comment by szhsin** [comment]
@fporzecki thanks for reporting the bug. I think your suggested fix makes sense and will solve the problem. We can add it into the next release.
Also, if you wish to do the fix, please feel free to submit a PR.

**Comment by fporzecki** [comment]
@szhsin I went ahead and created a PR with the exact fix I described

**Comment by szhsin** [comment]
Fixed in v3.5.3
